In [0]:
from google.colab import drive
drive.mount('/content/drive')

from pathlib import Path
drive_path = Path("/content/drive/My Drive/__USM Course Notes, Etc/Prob n Stats II")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
cd '/content/drive/My Drive/__USM Course Notes, Etc/Prob n Stats II/chp8'

/content/drive/My Drive/__USM Course Notes, Etc/Prob n Stats II/chp8


In [0]:
ls # Ensure we are in the right directory

 8.1-10.png  'Assignment 2.ipynb'   helper.py   images/   __pycache__/


<center> ![Imgur](https://i.imgur.com/bpMJwLo.png)


# *(a) Define the test statistic and critical region for $\alpha = 0.05$. Sketch a figure showing the critical region*


# $ n < 30$ and  $\sigma$ is  unkown, so we use Student T  distribution

$\large test\ statistic\ t = \frac{\bar{X} - \mu_{0}}{\frac{s}{ \sqrt{n}}}$

$\large critical\ region:\ t_{1 - \alpha}(n - 1)$

In [0]:
from helper import *

m0 = 5.70
n = 8
alpha = .05

criticalRegion = scipy.stats.t.ppf(1 - alpha, n - 1)
print(f"Lower Bound of Critical Region: {criticalRegion} ≈ {criticalRegion:.4}")

criticalRegion = float(round(criticalRegion, 3))
print(f"Critical Region: [{criticalRegion}, {math.inf})")


Lower Bound of Critical Region: 1.894578605061305 ≈ 1.895
Critical Region: [1.895, inf)


---
# *(b) Calculate the value of the Test Statistic and support your claim from part (a)*

In [0]:
# Load in sample data from problem
samples = load_sample_data(8,1,10, display=True)

# Calculate sample mean, sample variance and test statistic
xBar = sample_mean(samples)
s = sample_variance(samples) ** .5
m0 = 5.70
n = len(samples)
testStat = (xBar - m0) / (s / n ** .5)

# Display values
print()
print("=="*15)
print(f"Sample Mean: {xBar}  ≈  {xBar:.5}")
print(f"Sample Standard Deviation: {s}  ≈ {s:.4}")
print(f"Test Statistic: {testStat}  ≈  {testStat:.5}")

# Round to the nearest thousanth
xBar = round(xBar, 4)
s = round(s, 4)
testStat = round(testStat, 4)

# xBar, s, testStat, criticalRegion 

Loaded E8_1-10.txt sucessfully.
[5.71 5.8  6.03 5.87 6.22 5.92 5.57 5.83]

Sample Mean: 5.86875  ≈  5.8688
Sample Standard Deviation: 0.19737111526983148  ≈ 0.1974
Test Statistic: 2.4182721805487803  ≈  2.4183


<center> $\large t \approx 2.4183\  and\ is\ within\ rejection\ region\  [1.8946, \infty) \\  \large 1.895 < 2.4183 < \infty$
# *Therefore we reject $H_0$*

---

# *(c) Approximate the p-value and provide its bounds*

### *The p-value is the probability of obtaining the value of the test statistic or greater(or less than if negative). We use the Student T distribution to approximate this probability within the respected degrees of freedom.*

<center>$\Large P(\ t(n-1) \geq 2.183) \\ \Large = P(\ t(n-1) > 2.183) \\ \Large = 1 - P(\ t(n-1) \leq 2.183) \\ \Large = 1 - P(\ t(n-1) < 2.183)$

In [0]:
pVal = 1 - scipy.stats.t.cdf(testStat, n - 1)
print(f"P-value: {pVal}  ≈  {pVal:.4}")
pVal = round(pVal,4)

P-value: 0.023103679733782467  ≈  0.0231


---
<center> ![Imgur](https://i.imgur.com/hNsdZt9.png =500x) </center>

In [0]:
mD = 0
n = 17
alpha = .05

# Load the data sample data from text file
# Reshape array and 
samples = load_sample_data(8,1,12, True)
samples = np.reshape(samples, (17, 3)).T

# Split the data accordingly,
# x = Distance Ball A
# y = Distance Ball B
# d = Correct difference ( dist A - dist B)
x, y, d = (samples[0]), (samples[1]), (samples[2]) 

Loaded E8_1-12.txt sucessfully.
[265 252  13 272 276  -4 246 243   3 260 246  14 274 275  -1 263 246  17
 255 244  11 258 245  13 276 259  17 274 260  14 274 267   7 269 267   2
 244 251  -7 212 222 -10 235 235   0 254 255  -1 224 231  -7]


## <center> *We are performing a paired t-test. Must take the difference to find distribution D.*
 <center> $\large d_1 = x_1 - y_1, ... , d_n = x_n - y_n$

  ## <center> *The formula for Test Statistic t looks awfully familar*
  <center> $\large test\ statistic\ t = \frac{\bar{d} - \mu_d}{\frac{s_d}{\sqrt{n}}}$


In [0]:
# if difference is correct find sample mean/variance and test Statistic 
if (x - y).all() == d.all():
    dBar = sample_mean(d)
    dStd = sample_variance(d) ** .5
    testStatistic = (dBar - mD) / (dStd / n ** .5)

    # Print results
    print(f"Sample Mean: {dBar} ≈ {dBar:.4}")
    print(f"Sample Standard Deviation: {dStd} ≈ {dStd:.4}")
    print(f"Test Statistic = {testStatistic} ≈  {testStatistic:.4}")
    
    # Round to approximations
    dBar = round(dBar, 3)
    dStd = round(dStd, 3)
    testStatistic = round(testStatistic, 4)
# dBar, dStd, testStatistic

Sample Mean: 4.764705882352941 ≈ 4.765
Sample Standard Deviation: 9.086593226869367 ≈ 9.087
Test Statistic = 2.162018826797583 ≈  2.162


#### <center>*We need to compare test statistic t to $t_{1-\alpha}(n-1)$, the critical region* 

In [0]:
criticalRegion = scipy.stats.t.ppf(1 - alpha, n - 1)
print(f"Lower bound of Critical Region: {criticalRegion} ≈ {criticalRegion:.4}")
print(f"Critical Region: [{criticalRegion:.4}, {math.inf})")
criticalRegion = round(criticalRegion, 3)

Lower bound of Critical Region: 1.74588367627624 ≈ 1.746
Critical Region: [1.746, inf)


<center> $\large 2.162\ is\ within\ critical\ region\ [1.746, \infty) \\ \Large Therefore\ we\ reject\ H_0$

---
<center> ![Imgur](https://i.imgur.com/JK2rWwN.png?1 =500x) </center>

# *(a) Define the test statistic and critical region assuming the variances are equal*
## <center> *Subcase II.B.I : Assume $\sigma_{x}^2 = \sigma_{y}^2$*


</br></br>

|$\LARGE H_1$| $\LARGE We\ reject\ H_0\ if$|
|----|:-----------------------:-|
| $\Large \mu_x > \mu_y$ | <center> $  t \geq t_{\alpha}(n + m -2)$|
| $\Large \mu_x < \mu_y$ | <center>$ t \leq - t_{\alpha}(n + m -2)$|
| $\Large \mu_x \neq \mu_y$ | <center>$ |t| \geq t_{\frac{\alpha}2}(n + m -2)$|


<center>
  
</br></br>

### *$ n < 30$ use Student T distribution FOOL!*
$\Large test\ statistic\ t = \frac{\bar{X} - \bar{Y} }{S_{p} \sqrt{\frac{1}n + \frac{1} m}} \\ \small And\ S_p = \sqrt{ \frac{(n-1)S_x^2 + (m -1)S_y^2}{n + m - 2} }$

### *The lowerbound of our Critical region* **in this case $H_1 = (\mu_X < \mu_Y)$:**  
$\Large  -t_{\alpha}(n + m -2) \\ \Large = -t_{.05}(13 + 16 - 2)$


In [0]:
alpha = .05
n = 13
m = 16

# Find Critical Region, as defined above
criticalRegion = scipy.stats.t.ppf(alpha, n + m - 2)

print(f"Lower bound of Critical Region: {criticalRegion}  ≈ {criticalRegion:.4}")
print(f"Critical Region: [{criticalRegion:.4}, {-math.inf})")

# Round accordingly
criticalRegion = round(criticalRegion, 3)
# criticalRegion # print to check rounding

Lower bound of Critical Region: -1.703288445722127  ≈ -1.703
Critical Region: [-1.703, -inf)


# *(b) Given the values defined, Calculate the test statistic and state the conclusion on $H_0$*

<center>
$\Large test\ statistic\ t = \frac{\bar{X} - \bar{Y} }{S_{p} \sqrt{\frac{1}n + \frac{1} m}} \\ \small And\ S_p = \sqrt{ \frac{(n-1)S_x^2 + (m -1)S_y^2}{n + m - 2} }$
</center>

In [0]:
# Defined values
xBar = 72.9 
xStd = 25.6
yBar = 81.7
yStd = 28.3

# Calculate test stat
sP = (((n-1) * xStd ** 2 + (m - 1) * yStd **2 )/ (n + m - 2)) ** .5
testStat = (xBar - yBar) / (sP * (1/n + 1/m) ** .5)

# Display Result and round accordingly
print(f"Pooled Standard Derivation: {sP} ≈ {sP:.4}")
print(f"Test Statistic t: {testStat} ≈ {testStat:.3}")
testStat = round(testStat, 3)

print(f"Is the test stat within rejection region: {testStat < criticalRegion}") 
# sP, testStat

Pooled Standard Derivation: 27.133190007811468 ≈ 27.13
Test Statistic t: -0.8685893190338824 ≈ -0.869
Is the test stat within rejection region: False


<center> 
 ### We fail to reject $H_0$ is not within reject region  $[-1.703, \infty)$

# *(c) Give the bounds for the p-value of this test*

In [0]:
pBound = scipy.stats.t.cdf(testStat, n + m -2)
print(f"P-value: {pBound} ≈ {pBound:.5}")

testStat, (n + m -2)

P-value: 0.19625330562443638 ≈ 0.19625


(-0.869, 27)

### *Formally given our p-value is approximately .19625, however we were asked a the boundary of our p-value. To find this we use our test statistic t and degrees of freedom as reference when looking at the t chart </br></br> Basically we are looking for the probability that we within the area between our test statistic and  critical region The Area between our test stat and critical region: *

| Degrees of Freedom| $T_{.50}$| $T_{.25}$| $T_{.20}$|  $\textbf T_{.15} $| $\textbf  T_{.10}$| $T_{.05}$|
|---|----|----|----|----|----|----|
|27| 0.00|0.684 | 8.55|**1.057** | **1.314** | 1.703 |

</br>

<center> $ \LARGE   P(T_{.05} < P < T_{.20}) $ 


  
  </br>
 ![Imgur](https://i.imgur.com/4sbsItJ.png)
</center>

----
<center> ![Imgur](https://i.imgur.com/mGex54J.png =500x)  


## *(a) Test Null Hypothesis against one-sided alternative $\mu_x > \mu_y$*
###<center>  *Subcase II.B.II : Cannot Assume $\sigma_{x}^2 = \sigma_{y}^2$*
</br>

|$\LARGE H_1$| $\LARGE We\ reject\ H_0\ if$|
|----|:-----------------------:-|
| $\Large \mu_x > \mu_y$ | <center> $  t \geq t_{\alpha}(r)$|
| $\Large \mu_x < \mu_y$ | <center>$ t \leq - t_{\alpha}(r)$|
| $\Large \mu_x \neq \mu_y$ |<center> $ |t| \geq t_{\frac{\alpha}2}(r)$|


<center>
   $ \Large t = \frac{\bar{X} - \bar{Y} }{\sqrt{ \frac{S^2_{x}}n +\frac{S^2_{y}}m}} $  *,* 
  $\Large r = \frac{( \frac{S^2_{x}}n +\frac{S^2_{y}}m)^2}{ \frac{1}{n-1}(\frac{S^2_{x}}n)^2 +\frac{1}{m-1}(\frac{S^2_{y}}m)^2 } $

In [0]:
# Load and split the data accordingly
samples = load_sample_data(8,2,14)
samples = np.reshape(samples, (19, 2)).T
x, y = (samples[0]), (samples[1]) 

m = n = len(x)
m, n
alpha = .05 

# Calculate Sample mean, std and test Statistic
xBar, xStd = sample_mean(x), sample_variance(x) ** .5
yBar, yStd = sample_mean(y), sample_variance(y) ** .5
testStatistic = (xBar - yBar) / ( (xStd ** 2)/n + (yStd **2) /m) ** .5

# Print results
print("===" * 10)
print(f"X Sample Mean: {xBar} ≈ {xBar:.4} \nX Sample Standard Deviation: {xStd} ≈ {xStd:.4}")
print(f"Y Sample Mean: {yBar} ≈ {yBar:.4} \nY Sample Standard Deviation: {yStd} ≈ {yStd:.4}")
print(f"\nTest Statistic = {testStatistic} ≈  {testStatistic:.4}")
print("\n")

# Implement Welches T test at r d.o.f
r = (((xStd ** 2)/n + (yStd **2)/m) **2) / ( ((1/(n -1))* ((xStd ** 2)/n)** 2) + ((1/(m - 1))* ((yStd ** 2)/m) **2) )
criticalRegion = scipy.stats.t.ppf(1 - alpha, r)

print(f"Lower bound of Critical Region: {criticalRegion} ≈ {criticalRegion:.5}")
print(f"Critical Region: [{criticalRegion:.5}, {math.inf})")
criticalRegion = round(criticalRegion, 4)

print(f"\nIs the test stat within rejection region: {testStat > criticalRegion}") 

Loaded E8_2-14.txt sucessfully.
X Sample Mean: 5.994736842105263 ≈ 5.995 
X Sample Standard Deviation: 2.453393043705102 ≈ 2.453
Y Sample Mean: 4.3921052631578945 ≈ 4.392 
Y Sample Standard Deviation: 1.4062621019622212 ≈ 1.406

Test Statistic = 2.4703290690886788 ≈  2.47


Lower bound of Critical Region: 1.6997617211723555 ≈ 1.6998
Critical Region: [1.6998, inf)

Is the test stat within rejection region: False


### <center> *We fail to reject $H_0$ for  $t \ngeq t_{\alpha}(r)$*
  
  
## *(b) Contstruct 2 boxplots to confirm (a)*



In [0]:
minn = np.min(x)
median = np.median(x)
mx = np.max(x)
qr1 = np.median(x[:13])
qr3 = np.median(x[15:])
IQR = (qr3 - qr1) 
low_fence = qr1 - (1.5  * IQR)
up_fence = qr3 + (1.5 * IQR)

print("X's Five Number Summary")
print("="*20)
print(f"Min: {minn}, \nFirst Quartile: {qr1}, \nMedian: {median}, \nThird Quartile: {qr3}, \nMaximum: {mx}")
print(f"InterQuartile Range: {IQR:.5}")
print(f"Lower Fence: {low_fence}")
print(f"Upper Fence: {up_fence:.5}")

minn = np.min(y)
median = np.median(y)
mx = np.max(y)
qr1 = np.median(y[:13])
qr3 = np.median(y[15:])
IQR = (qr3 - qr1) 
low_fence = qr1 - (1.5  * IQR)
up_fence = qr3 + (1.5 * IQR)

print("\nY's Five Number Summary")
print("="*20)
print(f"Min: {minn}, \nFirst Quartile: {qr1}, \nMedian: {median}, \nThird Quartile: {qr3}, \nMaximum: {mx}")
print(f"InterQuartile Range: {IQR:.5}")
print(f"Lower Fence: {low_fence}")
print(f"Upper Fence: {up_fence}")



X's Five Number Summary
Min: 2.7, 
First Quartile: 6.15, 
Median: 6.1, 
Third Quartile: 4.5, 
Maximum: 11.4
InterQuartile Range: -1.65
Lower Fence: 8.625
Upper Fence: 2.025

Y's Five Number Summary
Min: 2.1, 
First Quartile: 5.0, 
Median: 4.8, 
Third Quartile: 4.125, 
Maximum: 6.45
InterQuartile Range: -0.875
Lower Fence: 6.3125
Upper Fence: 2.8125


<center> ![Imgur](https://i.imgur.com/DJcVvWJ.png)

----
<center> ![Imgur](https://i.imgur.com/BE7snIf.png?1 =500x)


## *(a) Which hypothesis would you accept if the significance level is $\alpha = 0.05$?*



|$\LARGE H_1$| $\LARGE We\ reject\ H_0\ if$|
|----|:-----------------------:-|
| $\Large p > p_0$ | <center> $\Large  z \geq z_{\alpha}$|
| $\Large p < p_0$ | <center>$\Large z\leq -z_{\alpha}$|
| $\Large p \neq p_0$ |<center> $\Large |z| \geq z_{\frac{\alpha}2}$|

<center>
  $\Large \hat{p} = \frac{x}n, \Large z = \frac{\hat{p} - p_0}{\sqrt{ \frac{p_0 (1-p_0)}n}}$

In [0]:
p = .75
x = 273
n = 390
alpha = .05

pHat = x/n 
print(f"P-hat: {pHat}")


testStatistic = (pHat - p) / ( (p * (1 - p))/n) ** .5
print(f"Test Statistic: {testStatistic:.3}")

criticalRegion = scipy.stats.norm.ppf(alpha)
print(f"Critical Region: [{criticalRegion:.4}, {-math.inf})")
print(f"\nIs the test stat within rejection region: {testStat > criticalRegion}") 

P-hat: 0.7
Test Statistic: -2.28
Critical Region: [-1.645, -inf)

Is the test stat within rejection region: True


### *<center> We reject $H_0$; $z < z_{\alpha}; (-2.28 \leq -.1645)$* 
  
## *(b) Which hypothesis would you accept if the significance level is $\alpha = 0.01$?*

In [0]:
alpha = .01
testStatistic = (pHat - p) / ( (p * (1 - p))/n) ** .5
print(f"Test Statistic: {testStatistic:.3}")

criticalRegion = scipy.stats.norm.ppf(alpha)
print(f"Critical Region: [{criticalRegion:.4}, {-math.inf})")
print(f"\nIs the test stat within rejection region: {testStat < criticalRegion}") 

Test Statistic: -2.28
Critical Region: [-2.326, -inf)

Is the test stat within rejection region: False


### *<center> We fail to reject $H_0$; $z \geq z_{\alpha}; -2.28 \geq -2.326$* 
  
##*(c) Find the p-value for this test?*
  
<center>
  $\Large Find\ P = P(Z < -2.28) \\\Large  = P(Z > 2.28) \\\Large= 1 - P(Z < 2.28)$

  
  </br>
  ![Imgur](https://i.imgur.com/f6qSoyL.png )
</center>

In [0]:
pVal = 1 - scipy.stats.norm.cdf(2.28)
print(f"P value: {pVal:.5}")

P value: 0.011304
